### PR TITLE
Added dotnetframework pack and component with dependency cookbooks c…

### PR DESCRIPTION
**Pull request by sahebjade**
…hocolatey, windows, chef_handler

### CODE REVIEW

File: packs/dotnet-framework.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+include_pack "custom"
```
Comment (schwankl) [inline packs/dotnet-framework.rb]: you probably want to extend genericlb instead or have 2 packs - dotnet-lb and dotnet-rrdns

File: packs/dotnet-framework.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  },
+  :attributes   => {
+    "chocolatey_package_source" => 'https://chocolatey.org/api/v2/'
+  }
+
+resource "secgroup",
+  :cookbook => "oneops.1.secgroup",
+  :design => true,
+  :attributes => {
+    "inbound" => '[ "22 22 tcp 0.0.0.0/0", "3389 3389 tcp 0.0.0.0/0" ]'
```
Comment (schwankl) [inline packs/dotnet-framework.rb]: whats 3389 for?

File: components/cookbooks/dotnetframework/recipes/install.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  CHEF_VERSION_EXPRESSION = /^Chef:\s(?<major>\d\d)\.(?<minor>\d{1}+)\.(?<patch>\d{1}+)$/
+  capture = CHEF_VERSION_EXPRESSION.match(`chef-client -v`.chomp)
+
+  if capture['major'].to_i >= 12 && capture['minor'].to_i >= 7
+
+    dotnet_version_package_name = JSON.parse(node.dotnetframework.dotnet_version_package_name)
+    dotnet_version_package_name.each do |dotnet_version,package_name|
+      Chef::Log.info("installing #{dotnet_version}")
+      chocolatey_package package_name do
+        source node.dotnetframework.chocolatey_package_source
```
Comment (schwankl) [inline components/cookbooks/dotnetframework/recipes/install.rb]: use mirror cloud service if it exists for choco ... this as is will not work most internal clouds

File: components/cookbooks/dotnetframework/metadata.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name             'Dotnetframework'
+maintainer       'Oneops'
+maintainer_email 'support@oneops.com'
+license          'Apache License, Version 2.0'
+description      'This cookbook install .net frameworks'
+long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
```
Comment (schwankl) [inline components/cookbooks/dotnetframework/metadata.rb]: remove this long_description ... we do not rsync md or png files and will cause runtime issue

File: packs/dotnet-framework.rb (hunk 1 above)
Comment (sahebjade) [inline packs/dotnet-framework.rb]: Thanks for the suggestion, I will extend genericlb

File: packs/dotnet-framework.rb (hunk 2 above)
Comment (sahebjade) [inline packs/dotnet-framework.rb]: 3389 is for remote desktop (rdp) for windows machine.

File: components/cookbooks/dotnetframework/recipes/install.rb (hunk 3 above)
Comment (sahebjade) [inline components/cookbooks/dotnetframework/recipes/install.rb]: At OS level chocolatey is being installed.

File: components/cookbooks/dotnetframework/metadata.rb (hunk 4 above)
Comment (sahebjade) [inline components/cookbooks/dotnetframework/metadata.rb]: sure I will remove the long_description

File: components/cookbooks/dotnetframework/recipes/install.rb (hunk 3 above)
Comment (schwankl) [inline components/cookbooks/dotnetframework/recipes/install.rb]: chocolatey_package_source will be https://chocolatey.org/api/v2/  ... who updates chocolatey_package_source to some internal repo?

**Pull request merged**